### ISSUE TITLE
adjust spacing between buttons in navigation bar?

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/joist#279]
In https://github.com/phetsims/joist/issues/241, I made it easy to adjust the spacing between the screen buttons in the navigation bar, as well as the spacing to the left of the home button.  See NavigationBar, constants `SCREEN_BUTTON_SPACING` and `HOME_BUTTON_LEFT_MARGIN`.  

If anyone thinks these values need tweaking, feel free to self assign. The work will be in reaching consensus.

**Comment by pixelzoom** [comment]
If this is going to be addressed, it should be done before redeploys.

Assigning to @ariel-phet to decide whether the current spacing is OK, and whether to pursue this issue. It's not an issue that I'm interested in tackling, but I've made it trivial to adjust the spacing.

**Comment by ariel-phet** [comment]
I think no tweaking is currently necessary. I do think it is good that we now have this flexibility just in case a truly long name feels necessary for a particular sim, or other constraints require such tweaking. Considering these spacings were all tweaked long ago in the initial design of the Nav bar and seem to have been working well, I see no reason to reopen the discussion.

**Comment by samreid** [comment]
When I was testing #278 I launched Hooke's Law and the navbar screen buttons seemed very crowded:

![image](https://cloud.githubusercontent.com/assets/679486/8749780/025c69b4-2c61-11e5-9ea5-925645af3d8f.png)

I wanted to make sure @ariel-phet took a close look at this--it wasn't clear from the above comments whether this was intentionally changed, or whether @ariel-phet is aware of the change.  I looked into a previous version of Hooke's Law to see if there was really a change, and it looks like there was--here is hookes-law 1.0.0-dev.21:

![image](https://cloud.githubusercontent.com/assets/679486/8749887/7b292030-2c61-11e5-9bac-594c8654680f.png)

So it seems like this navbar screen button crowding is a new issue, and I'd like @ariel-phet to take a close look and make a recommendation.

**Comment by ariel-phet** [comment]
@samreid - I was not aware, my recommendation to @pixelzoom was basically to keep whatever we have used in the past as the default spacing.

Can you fix this issue?

**Comment by samreid** [comment]
Sure, I'll take a look.

**Comment by samreid** [comment]
By the way, this doesn't look like a problem for the *English* translation of Molecule Shapes, but it could  affect the translations if their text is shorter.  Not sure if I'll get to this before Monday's MS deploy or not.

**Comment by samreid** [comment]
When I made the text very small (such as stringTest=x) and left the spacing as 0, the buttons still had some space between them:

![image](https://cloud.githubusercontent.com/assets/679486/8759734/221508b2-2cb9-11e5-9c8a-e31a75ec203c.png)

I think this is because each NavigationBarScreenButton has invisible width where the highlight will appear.  Perhaps solving this problem will be as simple as adjusting SCREEN_BUTTON_SPACING?

**Comment by samreid** [comment]
By adjusting SCREEN_BUTTON_SPACING and HOME_BUTTON_LEFT_MARGIN and comparing hookes-law master with hookes-law 1.0.0-dev.21, I determined that a value of 11 for both of these constants matches the previous version.  However, these tests were only in the isolated case of "3 screens with short names" and we should check to see if any of the other combinations are negatively impacted.  It would also be good to hear from @pixelzoom why he chose these values of 0 and 0 once he returns from his vacation.

**Comment by samreid** [comment]
Using 11 for the constants mentioned above pushes the icons further apart than in the published 1.0.0 of Molecule Shapes:

master with 11/11:
![image](https://cloud.githubusercontent.com/assets/679486/8763634/376e6658-2d64-11e5-8341-89f63fa7c114.png)

1.0.0

![image](https://cloud.githubusercontent.com/assets/679486/8763632/280f15fe-2d64-11e5-83ea-4e551e0c6df0.png)

In this case, the latter looks better.  I think the issue is that all of the buttons have the same width, and if one of the buttons has wider text than the other, the leftover space for one of the buttons automatically functions as "spacing", so adding further spacing seems like too much.

**Comment by samreid** [comment]
I think what we actually want to set (and make sure exceeds a minimum) is the spacing between the *icons*, not the buttons (which includes text and highlight padding).

**Comment by samreid** [comment]
If I add this line of code after the declaration of maxScreenButtonWidth

```js
maxScreenButtonWidth = Math.max( maxScreenButtonWidth, 60 );
```

It gets pretty close to a good spacing for the "short text" case, and doesn't disrput the larger text case.  Together with `var HOME_BUTTON_LEFT_MARGIN = 8;` this does a pretty good job spacing things and matching previous convention.  The exception is that for the "short text" case, the home button is slightly closer than normal:

original:
![image](https://cloud.githubusercontent.com/assets/679486/8763703/d13f52b2-2d67-11e5-8bbe-77ddfa520252.png)

with settings as described in this issue comment:
![image](https://cloud.githubusercontent.com/assets/679486/8763707/ea29e30a-2d67-11e5-94dc-203545035696.png)

**Comment by samreid** [comment]
Since @jessegreenberg reviewed #241 it would be good to get his opinion on the workaround proposed in https://github.com/phetsims/joist/issues/279#issuecomment-122601724

Particularly: (a) do you think this will cause complications for any of the other combinations and (b) do you have any other suggestions how to solve this problem?

**Comment by jessegreenberg** [comment]
> It would also be good to hear from @pixelzoom why he chose these values of 0 and 0 once he returns.

During the review of #241, @pixelzoom said that 

>I set it back to zero because the additional spacing (requested by @samreid) felt too different from what we've had in the past.

I think values were set to zero for cases like Molecule Shapes shown above.

> I think the issue is that all of the buttons have the same width, and if one of the buttons has wider text than the other, the leftover space for one of the buttons automatically functions as "spacing", so adding further spacing seems like too much.

After #241, all of the buttons have different widths, but the spacing is still set by the button with the max width so I agree that this is the issue.

**Comment by jessegreenberg** [comment]
I don't see any issues with the workaround suggested by @samreid, and it was used in NavigationBar before #241.
```
var maxScreenButtonWidth = Math.max( 50, _.max( screenButtons, function( button ) { return button.width; } ).width );
```
The value 50 was used because NavigationBarScreenButton used to have its own horizontal padding.  @pixelzoom removed this workaround in commit 30940a0137f9c40d68ec43b2a4c6deb446838168 because 50 was an undocumented magic number. If we want to match the original layout, I think that
```
var maxScreenButtonWidth = Math.max( 60, _.max( screenButtons, function( button ) { return button.width; } ).width );
```
with HOME_BUTTON_LEFT_MARGIN = 8 would solve the issue for all cases illustrated in above comments.

**Comment by samreid** [comment]
I implemented the suggestions from the above comment, and tested hookes-law, forces-and-motion-basics and acid-base-solutions.  Everything seemed acceptable, so I'll go ahead and commit it.  Thanks for discussing this with me @jessegreenberg.

**Comment by samreid** [comment]
I pushed the changes above.  It is a small commit, but can you review it @jessegreenberg just to make sure everything is OK since we have imminent deploys from master coming?  

Also, today is Molecule Shapes day, so I decided to test Molecule Shapes and compare the production version to master.  Here is what I saw:

![image](https://cloud.githubusercontent.com/assets/679486/8779440/b80fb6b0-2ec0-11e5-86fd-220c91083fd4.png)

The size and spacing is pretty much the same, but master (the bottom pic) has everything translated to the left a noticeable amount.  For sims with bilateral symmetry, it is nice to have the screen buttons perfectly centered in the navbar, so I'll look into this and see which is accurate (if either is).

**Comment by samreid** [comment]
To determine which (if any) of the above navbar screen icon buttons is centered, I drew a line down the middle of the above image:

![centered](https://cloud.githubusercontent.com/assets/679486/8779688/c78c3d9c-2ec1-11e5-8115-59fedf5c5b9c.png)

Based on the above drawing, it looks like the production version of molecule shapes is centered, and the master version is shifted to the left.  While related, this seems like a separate problem so I will create a new issue for it.  I'll also mark it as S2015R, though for sims without bilateral symmetry such as Molecule Shapes, it is not as important.

**Comment by samreid** [comment]
Reassigned to @jessegreenberg to review as requested in https://github.com/phetsims/joist/issues/279#issuecomment-122915342

**Comment by jessegreenberg** [comment]
It looks good @samreid, and thanks for adding the clear documentation.  I think that is why it was deleted the first time.  Closing since there is a new ticket for the issue described in https://github.com/phetsims/joist/issues/279#issuecomment-122916990
